### PR TITLE
Extract trace id generator in an interface to allow different implementations

### PR DESCRIPTION
**Pull request by bogdandrutu**
Currently the mechanism to inject a different ids generator is not implemented, but soon we will implement for clock, resource, etc.

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/trace/IdsGenerator.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+public interface IdsGenerator {
+  /**
+   * Generates a new random {@code SpanId}.
+   *
+   * @return a valid new {@code SpanId}.
+   */
+  SpanId generateSpanId();
+
+  /**
+   * Generates a new random {@code TraceIde}.
```
Comment (breedx-nr) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/IdsGenerator.java]: typo: `TraceIde` -> `TraceId` (so hard to not type the `e`...I feel ya 😄 )

File: sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      id = random.nextLong();
+    } while (id == INVALID_ID);
+    return new SpanId(id);
+  }
+
+  @Override
+  public TraceId generateTraceId() {
+    long idHi;
+    long idLo;
+    do {
```
Comment (breedx-nr) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java]: Recommend encapsulating the `do { ... } while (id is invalid)` logic into a method and leveraging it from both generate methods.

Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java]: https://github.com/open-telemetry/opentelemetry-java/issues/642

File: sdk/src/main/java/io/opentelemetry/sdk/trace/IdsGenerator.java (hunk 1 above)
Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/IdsGenerator.java]: Done.

**Pull request merged**